ad4m-hooks/vue/src/useModel.ts

```diff
@@ -0,0 +1,132 @@
+import { ref, Ref, watch, computed, shallowRef } from "vue";
+import { PerspectiveProxy, Ad4mModel, Query, PaginationResult } from "@coasys/ad4m";
+
+type Props<T extends Ad4mModel> = {
+  perspective: PerspectiveProxy | any;
+  model: string | ((new (...args: any[]) => T) & typeof Ad4mModel);
+  query?: Query;
+  pageSize?: number;
+  preserveReferences?: boolean;
+};
+
+type Result<T extends Ad4mModel> = {
+  entries: Ref<T[]>;
+  loading: Ref<boolean>;
+  error: Ref<string>;
+  totalCount: Ref<number>;
+  loadMore: () => void;
+};
+
+export function useAd4mModel<T extends Ad4mModel>(props: Props<T>): Result<T> {
```

Comment:
Name inconsistent with react hook. should both be:
```suggestion
export function useModel<T extends Ad4mModel>(props: Props<T>): Result<T> {
```